Fixing DDR preset issues

DobbyDigital opened the pull request:
Fixing some bugs that popped up with DDR preset when it was pushed to online Dev build. Some name change issues, and one preset issue.


- Modified DDR preset to use the new name for Ganon BK on LACS (did not crash previously, but incorrectly defaulted to Own Dungeon)
- item_hints did not load correctly from hint distro, so added a line in world.py to dump the appropriate items in there (Note: This includes two copies of every item in the spoiler log, and I have no idea why)
- Modified DDR hints file to include the items to be hinted (using the 'named-item' type)
- Added a null check in get_specific_item_hint in case the list is empty (which happened previously with the DDR preset, but could happen more generally)

DobbyDigital committed:
Patch implemented

- Modified DDR preset to use the new name for Ganon BK on LACS (did not crash previously, but incorrectly defaulted to Own Dungeon)
- item_hints did not load correctly from hint distro, so added a line in world.py to dump the appropriate items in there (Note: This includes two copies of every item in the spoiler log, and I have no idea why)
- Modified DDR hints file to include the items to be hinted (using the 'named-item' type)
- Added a null check in get_specific_item_hint in case the list is empty (which happened previously with the DDR preset, but could happen more generally)

mracsys commented on World.py:
```diff
@@ -159,6 +161,7 @@ def __init__(self, id, settings):
                     raise Exception('Custom hint text too large for %s', loc['location'])
                 self.hint_text_overrides.update({loc['location']: loc['text']})
 
+        self.item_hints+=self.item_added_hint_types["named-item"]
```
Recommend changing this to `self.item_hints = settings.item_hints + self.item_added_hint_types["named-item"]` to prevent the item duplication issue in the spoiler. This will also make the key in the spoiler match the plando file if used.

Zannick replied:
I was going to test this and suggest it this morning. Thanks!

Zannick replied:
Ah, this does wipe out the spoiler `item_hints` though. Is that okay?

DobbyDigital replied:
Since the `item_hints` list is pre-defined by the hint distribution anyway, it's not a big deal to not have access to it in the spoiler log. This also prevents any potential issues of feeding back the spoiler log to the generator as a plando file, so it works out.

The only nice thing having the item_hints in the spoiler log would do is tell you for certain which of the items in the list the generator picked to hint, though that isn't really a big deal to not always know .

Zannick replied:
Excellent then.

DobbyDigital committed:
Added spoiler log fix

Added the change recommended by mracsys. Verified, and it does indeed prevent the duplicated items in the spoiler log and in hint generation. Thanks mracsys! <3
